### ISSUE TITLE
 rancher authentication between karma and alertmanager

### ISSUE THREAD

**Issue by ncauchois** [prymitive/karma#2108]
I set up an authentication between karma and alertmanager via rancher.
It works very well for viewing alerts, but not for sending silences.

My configuration of karma :
```
alertmanager:
      interval: 30s
      servers:
        - name: ocb-test01
          uri: https://rancher.xxxx/k8s/clusters/local/api/v1/namespaces/monitoring/services/alertmanager-operated:9093/proxy/
          headers:
            Authorization: "Bearer xxxxx"
          timeout: 20s
          proxy: true
          tls:
            insecureSkipVerify: true

```

I tested with a curl and it works fine :
```
curl -X POST -H "Authorization: Bearer xxxx" https://rancher.xxxx/k8s/clusters/local/api/v1/namespaces/monitoring/services/alertmanager-operated:9093/proxy/api/v2/silences
```

Looking at the rancher logs I had the impression that the token is not sent.

**Comment by prymitive** [comment]
I've started nc in listen mode on port 9093:
```
$ nc -v -l 9093
```

Then I started karma with this config file:
```
alertmanager:
  interval: 10s
  servers:
    - cluster: HA
      name: ha1
      uri: "http://localhost:9093"
      timeout: 10s
      headers:
        Authorization: "Bearer xxxxx"
```

And I see that the header is being passed:
```
% nc -v -l 9093
GET /metrics HTTP/1.1
Host: localhost:9093
User-Agent: Go-http-client/1.1
Accept-Encoding: gzip
Authorization: Bearer xxxxx
```

**Comment by prymitive** [comment]
I've also used https://github.com/jmartin82/mmock to ensure that all requests have it set:

![screencapture-localhost-8082-2020-08-26-22_17_53](https://user-images.githubusercontent.com/114288/91358106-278ae900-e7ea-11ea-96fe-cd88f1ebaa01.png)

**Comment by ncauchois** [comment]
You test with get methods and it works fine. The problem I encounter is only with post methods and directly with karma.

**Comment by prymitive** [comment]
Ah, sorry, missed that.
I don't see any headers being set in proxy code, so that does look like a bug.

**Comment by prymitive** [comment]
Should be fixed via #2115

**Comment by ncauchois** [comment]
It works well with the fix, thank you.
